source/fab/steps/archive_objects.py

```diff
@@ -50,18 +74,35 @@ def run(self, artefact_store: Dict, config):
         """
         super().run(artefact_store, config)
 
-        output_fpath = Template(self.output_fpath).substitute(output=config.project_workspace / BUILD_OUTPUT)
-
-        compiled_files: List[CompiledFile] = self.source_getter(artefact_store)
-
-        command = [self.archiver]
-        command.extend(['cr', output_fpath])
-        command.extend(map(str, compiled_files))
-
-        log_or_dot(logger, 'CreateObjectArchive running command: ' + ' '.join(command))
-        try:
-            run_command(command)
-        except Exception as err:
-            raise Exception(f"error creating object archive: {err}")
-
-        artefact_store[self.output_collection] = [output_fpath]
+        # We're expecting one or more build targets in the artefact store.
+        # When building exes, each build target has a name and a list of compiled files.
+        # When building a shared object there is a single build target with no name.
```

Comment:
We haven't spent much time considering the use case for multiple libraries from a single source. It's not clear if there's call for it, or if it would be better to have a separate a build config per library. Always happy to revisit this, of course.

Reply:
We haven't thought much about this use case because it's not clear if there's a call for it, or if it would be better to have a separate build config for each library. Always happy to revisit this, of course.